async-nats/src/lib.rs

```diff
@@ -847,23 +876,69 @@ impl ConnectionHandler {
 /// Client is a `Clonable` handle to NATS connection.
 /// Client should not be created directly. Instead, one of two methods can be used:
 /// [connect] and [ConnectOptions::connect]
-#[derive(Clone)]
 pub struct Client {
     sender: mpsc::Sender<Command>,
+    errors: Option<mpsc::Receiver<String>>,
```

Comment:
Did you consider using a broadcast channels for this?

Reply:
I did. But to clone a broadcast receiver you have to call it on `Sender`, which would mean we have to store `Sender` only to be able to call `Clone`...

None looks good.

Reply:
@caspervonb so, there seem to be no clean way (with mpsc).
We can either do what I did, or also include `Sender` in `Client` just to be able to call `subscribe` which returns another copy of `receiver`.

Only other option is using `flume` or `crossbeam` for errors, which has capabilities of cloning receiver and which I also don't like.

Reply:
A bit convoluted way would be to decouple it completely and have a command that sends the sender to the dispatcher, returning the receiver for the user.

Client doesn't really care what you do with it nor contribute to the ownership of it.